python/cudf/cudf/core/column/string.py

```diff
@@ -411,13 +418,18 @@ def as_decimal_column(self, dtype: DecimalDtype) -> DecimalBaseColumn:
         return result  # type: ignore[return-value]
 
     def as_string_column(self, dtype) -> StringColumn:
+        col = self
         if dtype != self.dtype:
             if isinstance(dtype, pd.StringDtype) or (
                 isinstance(dtype, pd.ArrowDtype)
                 and pa.string() == dtype.pyarrow_dtype
             ):
-                self._dtype = dtype
-        return self
+                col = self.copy(deep=True)
```

Comment:
If we're just modifying the `._dtype` attribute can't we shallow copy instead?

Reply:
In quite a few places of our code-base we do `col.astype('str'/'object')` where `col` isn't actually an intermediary and performing an `astype` with `np-dtype -> extension-dtype` will need to create a copy rather than modifying the first `np-dtype` column's metadata.

Reply:
OK thanks for the explanation. Yeah definitely not a blocker but would be good to not have to deep copy just to change dtype metadata in the future

Reply:
OK thanks for the explanation. Yeah definitely not a blocker but would be good to not have to deep copy just to change dtype metadata in the future